framework/include/base/MeshGeneratorSystem.h

```diff
@@ -0,0 +1,199 @@
+   *
+   * Sub generators are also generated within this phase, although the
+   * dependency resolution described above is only for the dependencies
+   * that we can parse using InputParamters. However, we require that
+   * sub generators be constructed within _their_ dependency order
+   * (except for the last one, which may be coupled to via the generator
+   * creating said sub generator).
+   *
+   * Should only be called by the CreateAddedMeshGenerator action during
+   * the create_added_mesh_generators task.
+   */
+  void createAddedMeshGenerators();
+
+  /**
+   * Get names of all mesh generators
+   * Note: This function should be called after all mesh generators are added with the
+   * 'add_mesh_generator' task. The returned value will be undefined and depends on the ordering
+   * that mesh generators are added by MOOSE if the function is called during the
+   * 'add_mesh_generator' task.
+   */
+  std::vector<std::string> getMeshGeneratorNames() const;
+
+  /**
+   * @returns Whether or not a mesh generator exists with the name \p name.
+   */
+  bool hasMeshGenerator(const MeshGeneratorName & name) const;
+
+  /**
+   * @returns The MeshGenerator with the name \p name.
+   */
+  const MeshGenerator & getMeshGenerator(const std::string & name) const;
+
+  /**
+   * Whether or not we know about the parameters for a MeshGenerator with the given name
+   *
+   * This is primarily for error checking. If MeshGenerator dependencies are screwed up,
+   * someone could be looking for a MeshGenerator that hasn't been constructed yet.
+   * With this, at least we can give the user some context that we know the generator
+   * exists, just that the dependencies are hosed.
+   */
+  bool hasMeshGeneratorParams(const MeshGeneratorName & name) const;
+
+  /**
+   * Get the generated mesh generated by executeMeshGenerators();
+   */
+  std::unique_ptr<MeshBase> getMeshGeneratorMesh(const bool check_unique = true);
+
+  /**
+   * Whether or not mesh generators are currently being appended (append_mesh_generator task)
+   */
+  bool appendingMeshGenerators() const;
+
+private:
+  /**
+   * Gets the MeshGeneratorNames that are referenced in an object's parameters.
+   *
+   * The result is returned as a pair of param name -> MeshGeneratorName in order
+   * to provide context.
+   *
+   * The \p allow_empty param sets whether or not to report names that are empty.
+   */
+  std::vector<std::pair<std::string, MeshGeneratorName>>
+  getMeshGeneratorParamDependencies(const InputParameters & params,
+                                    const bool allow_empty = false) const;
+
+  /**
+   * Order all of the _mesh_generators into _ordered_mesh_generators for later
+   * execution in executeMeshGenerators
+   */
+  void createMeshGeneratorOrder();
+
+  /**
+   * Internal method for actually constructing a mesh generator after it
+   * has been declared externally in addMeshGenerator (in Actions).
+   */
```

Comment:
Add a note about the ordering of mesh generator construction (name or dependency?)

Reply:
yeah note should go to   void createAddedMeshGenerators();
it s good there s something already